src/haxeLanguageServer/Context.hx

```diff
@@ -415,7 +426,8 @@ class Context {
 	}
 
 	function updateActiveEditorPackage(uri:DocumentUri):Void {
-		if (!uri.isHaxeFile()) {
+		// editor selection can happen before server initialization
+		if (determinePackage == null || !uri.isHaxeFile()) {
```

Comment:
What exactly does this fix? Not a fan of (ab)using `determinePackage` here either way, seems pretty random / hacky.

Reply:
`determinePackage` is null when workspace is opened and server has not started. I tried to extract package from active editor path, but this is complicated.

Reply:
Oh, I didn't see that `determinePackage ` is being used a few lines later here (wasn't shown without expanding the diff). Nvm, that makes more sense then.